pkg/config/node.go

```diff
@@ -487,7 +487,17 @@ func getNodeIpForRequestedIpStack(node v1.Node, filterIps []string, machineNetwo
 // statPort: The port on which the haproxy stats endpoint listens.
 // clusterLBConfig: A struct containing IPs for API, API-Int and Ingress LBs
 func GetConfig(kubeconfigPath, clusterConfigPath, resolvConfPath string, apiVips, ingressVips []net.IP, apiPort, lbPort, statPort uint16, clusterLBConfig ClusterLBConfig) (node Node, err error) {
-	if onPremPlatform, _ := isOnPremPlatform(clusterConfigPath); !onPremPlatform {
+	onPremPlatform := true
+	if clusterConfigPath != "" {
+		if onPremPlatform, err = isOnPremPlatform(clusterConfigPath); err != nil {
+			return Node{}, err
+		}
+	} else {
+		if len(clusterLBConfig.ApiIntLBIPs) > 0 {
+			onPremPlatform = false
+		}
+	}
+	if !onPremPlatform {
```

Comment:
I have to admit, I have a hard time understanding this logic. Correct me if I read it wrong:

- If clusterConfigPath is accessible, you determine the platform by calling isOnPremPlatform function
- If clusterConfigPath is NOT accessible, if at least 1 ApiIntLBIPs is provided, the platform is NOT on-prem

If this is what you wanted, why is that? What do I miss so that existence of ApiIntLBIPs is enough to say that you are NOT on-prem? What will happen if I take `platform: baremetal` with ApiIntLBIPs but without install-config?

Reply:
`BareMetal` and other on-prem platforms only sets the `api-vips` field. The `ApiIntLBIPs` are set by https://github.com/openshift/baremetal-runtimecfg/blob/main/cmd/runtimecfg/render.go#L37 only by cloud platforms so I felt confident to use that field to determine the platform type. 
I agree that approximation is not ideal.

So, now I have updated PR to bring in another parameter,  "--platform-type" to render. This parameter is currently passed in only when running on cloud platforms.